### PR TITLE
app-i18n/transifex-client-0.14.3: requires pip to run tests

### PR DESCRIPTION
**Pull request by robert7k**
Closes: https://bugs.gentoo.org/828780

### CODE REVIEW

**Comment by thesamesam** [comment]
I think it wants wheel, not pip. It's trying to use pip to install wheel.

**Comment by robert7k** [comment]
> It's trying to use pip to install wheel.

Nope, it's trying to run `pip wheel`, which requires both `pip` and `wheel`. I updated the commit accordingly.

**Comment by thesamesam** [comment]
> > It's trying to use pip to install wheel.
> 
> Nope, it's trying to run `pip wheel`, which requires both `pip` and `wheel`. I updated the commit accordingly.

I'm still not sure that's right. This is a common pattern (where pip will try to download something due to a missing dependency). It's calling `pip wheel` for it to satisfy another dependency:
```
/usr/bin/python3.9: No module named pip
error: Command '['/usr/bin/python3.9', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/var/tmp/portage/app-i18n/transifex-client-0.14.3/temp/tmpbpsaorod', '--quiet', 'smmap<5,>=3.0.1']' returned non-zero exit status 
```

It's asking `pip wheel` to install something satisfying `smmap<5,>=3.0.1`. There is no reason for it to be needing either `pip` or seemingly `wheel` when all dependencies are satisfied.

If I `emerge --ask --with-test-deps --onlydeps app-i18n/transifex-client`, then `ebuild transifex-client-0.14.3.ebuild clean test install`, it'll fail without smmap installed. It works if I install it without needing `pip` or `wheel`. The error is just confusingly warning about `pip` because it can't even call `pip` to _try_ to download smmap. With `pip` (and I guess `wheel`) installed, you'd get errors like this before it failed:
```
WARNING: The wheel package is not available.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba884f0100>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/smmap/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba884f0490>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/smmap/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba884f0160>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/smmap/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba88564cd0>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/smmap/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba88564d90>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/smmap/
ERROR: Could not find a version that satisfies the requirement smmap<6,>=3.0.1 (from versions: none)
ERROR: No matching distribution found for smmap<6,>=3.0.1
error: Command '['/usr/bin/python3.8', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/var/tmp/portage/app-i18n/transifex-client-0.14.3/temp/tmpkda8ebpx', '--quiet', 'smmap<6,>=3.0.1']' returned non-zero exit status 1.
 * ERROR: app-i18n/transifex-client-0.14.3::gentoo failed (test phase):
 *   Tests failed with python3.8
```

TL;DR: No `pip` or `wheel` needed, just smmp.

**Comment by robert7k** [comment]
> TL;DR: No `pip` or `wheel` needed, just `dev-python/smmap` (possibly with a version restriction).

Thanks for your analysis, I adjusted the patch accordingly.

I can confirm that it fails with `smmap`, `pip` and `wheel` uninstalled and works once `smmap`, but not `pip` and `wheel`, are installed.
